features/proposal_publication.feature

```diff
@@ -0,0 +1,29 @@
+Feature: Publishing and updating a proposal via Nostr
+  As a pillar in the HyperCore One ecosystem I want to be able to
+  publish proposals and update my published proposals.
+
+  Background:
+    Given the pillar "PillarOne" has a registered npub of "<NPUB>"
+    And the system is listening to Nostr events on configured relays
+
+  Scenario: Publishing a proposal
+    When a proposal event is received with:
+      | id            | event_number | title | description | minimum_znn | minimum_qsr | status | owner  | owner_signature |
+      | <PROPOSAL_ID> | 1            | Title | Description | 1000        | 10000       | 0      | <NPUB> | <SIG_VALID>     |
+    And the ID is a SHA-256 hash created from the serialized data of event_number, title, description, minimum_znn, minimum_qsr, and owner
+    And no proposal exists with an ID of "<PROPOSAL_ID>"
+    Then a new proposal is created
+
+  Scenario: Updating a proposal
+    When a proposal event is received with:
+      | id            | event_number | title | description | minimum_znn | minimum_qsr | status | owner  | owner_signature |
+      | <PROPOSAL_ID> | 2            | Title | Description | 1000        | 10000       | 0      | <NPUB> | <SIG_VALID>     |
+    And a proposal exists with an ID of "<PROPOSAL_ID>"
```

Comment:
the ID is defined as sha256(event_number, …), but the update scenario reuses the same <PROPOSAL_ID> with event_number=2. Isn't this contradictory?

Reply:
the hash of the content is just used to generate a unique ID
the ID can then be fixed while the contents change

this pattern is consistent with NoM where the creating tx hash is used as the id for zts tokens, staking entries, fusions, etcs. For example, iirc when you create a pillar you set an initial share percentages which influences the tx hash and thus pillar id. But when you change those percentages, the pillar ID doesn't change.

An alternative design for updates could be that they are basically new proposals (with new ids) with a pointer to the old proposal being replaced.